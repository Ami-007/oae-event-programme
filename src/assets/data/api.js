import axios from "axios";

export const getEvent = async (id) => {
  const url = `https://oae-mock-api.herokuapp.com/events/${id}`;

  return axios.get(url).then((event) => event.data);
};
