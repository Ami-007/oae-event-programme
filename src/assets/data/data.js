export const getMockEventById = (id) => {
  return events.filter((event) => event.id === id.toString())[0];
};

// MOCK DATA
export const events = [
  {
    id: "6bb8ffe8-6e5e-4abc-be7f-b05b749849cf",
    name: "Infant Musicality",
    series: "Bach, The Universe & Everything",
    time: "11.30",
    date: "12-12-21",
    venue: "King's Place",
    location: "London",
    imageSrc:
      "https://oae.co.uk/wp-content/uploads/2021/07/MicrosoftTeams-image-5-1250x825.jpg",
    imageAlt: "Event Poster",
    content: [
      {
        heading:
          "If this is your first Bach, the Universe & Everything (BAE), welcome!",
        text: "We like to think of the series as a community, similar to the one Bach enjoyed in Leipzig where he produced cantatas at an extraordinary rate, providing innovative music for the weekly services at the church where he worked from 1723 until his death in 1750. At that time, the congregation was as open to science and new ideas as it was united by faith. It is in the spirit that we come together for a thought-provoking cantata and a talk from a leading scientist. Professor Jon Chapman from the University of Oxford invites us to ponder the fleeting moment when sound waves reach our ears before they disappear. Using some careful calculations, Jon suggests that we might even be able to hear the shape of a drum.",
      },
      {
        heading:
          "A note from our Principal Keyboard, Steven Devine, about today's programme",
        text: "The Cantata being performed today is BWV 26 - Ach wie flüchtig, ach wie nichtig. The chorale of this name forms the linchpin of the outer movements and it is the only time Bach used this chorale in his cantatas. The melody of the chorale, written by Michael Franck in 1652, is used by others and the organ prelude today is a movement for manuals only (that is, no organ pedals!) from a set of variations on this tune by Georg Bohm (1661 - 1733), an organist whom J S Bach hugely admired and who may even have been his teacher.The irresistable combination of three oboes brought a wealth of different repertoire to mind, but I have been enjoying digging into lesser-known works by George-Philipp Telemann this season and this movement from his “Darmstadt” Ouverture (the title of which roughly translates as “boasting”!) is a witty interplay between the wind and string teams.",
      },
    ],
    schedule: [
      {
        name: "Welcome",
        title: null,
        author: null,
        description:
          "Non ad labore aliqua adipisicing et laborum exercitation. Enim consectetur aliquip eu commodo. Eiusmod ipsum anim exercitation consectetur elit amet voluptate aliquip ullamco. Duis deserunt dolore et officia ullamco voluptate culpa eiusmod cupidatat ullamco sit anim incididunt nulla. Nisi et et commodo non dolor amet aliquip exercitation. Et fugiat consequat mollit cillum anim aliquip commodo laboris enim voluptate ullamco ipsum nostrud qui. Laboris duis cillum duis aliqua cupidatat nulla.",
        alternateDescription:
          "Not to do any iterative fatigue and fatigue. For the main objectives of the EU. This type of exercise enhanced the mind's desire to enjoy the objectives of any school. They abandon the school district offices in pain and pleasure through no fault of that kind If not, the pain will not be comfortable and the objectives of the exercise are important. And he shuns the consequences of the softening of his soul, for the pleasure of his labors, for the pleasure of any one of us. Laboris duis cillum duis something longing for none.",
      },
      {
        name: "Cantata",
        title: "Ach wie fluchtig, ach wie nichtig BWV 26",
        author: "JS Bach",
        description:
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Id porta nibh venenatis cras sed felis eget velit.",
        alternateDescription:
          "The pain itself is the love of the pain, the main ecological problems, but I give this kind of time to fall down, so that some great pain and pain. That's the gate of the venenatis tomorrow, but the cat needs to choose.",
      },
    ],
    theme: {
      fontType: 1,
      primaryColour: "#8BA0E3",
      textColour: "#FFFFFF",
      highlightColour: "#FFFFC7",
    },
  },
  {
    id: "5101a840-5606-4097-9b66-a567213ae915",
    name: "Can You Hear The Shape of a Drum?",
    series: "Bach, The Universe & Everything",
    time: "11.30",
    date: "21-11-21",
    venue: "King's Place",
    location: "London",
    imageSrc:
      "https://oae.co.uk/wp-content/uploads/2021/07/Shape-of-drum-artwork-fb-event-1250x825.jpg",
    imageAlt: "Event Poster",
    featuredEvent: false,
    content: [
      {
        heading:
          "If this is your first Bach, the Universe & Everything (BAE), welcome!",
        text: "We like to think of the series as a community, similar to the one Bach enjoyed in Leipzig where he produced cantatas at an extraordinary rate, providing innovative music for the weekly services at the church where he worked from 1723 until his death in 1750. At that time, the congregation was as open to science and new ideas as it was united by faith. It is in the spirit that we come together for a thought-provoking cantata and a talk from a leading scientist. Professor Jon Chapman from the University of Oxford invites us to ponder the fleeting moment when sound waves reach our ears before they disappear. Using some careful calculations, Jon suggests that we might even be able to hear the shape of a drum.",
      },
      {
        heading: "Can you hear the shape of a Drum?",
        text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam sagittis augue et urna volutpat aliquam quis at ipsum. ",
      },
    ],
    schedule: [
      {
        name: "Welcome",
        title: null,
        author: null,
        description: null,
        alternateDescription: null,
      },
      {
        name: "Prelude",
        title: "Ach wie fluchtig, ach wie nichtig",
        author: "Georg Bohm",
        description: null,
        alternateDescription: null,
      },
      {
        name: "Polyphony",
        title: "Haec Dies",
        author: "William Byrd",
        description: null,
        alternateDescription: null,
      },
      {
        name: "Reading",
        title: "Beat! Beat! Drums!",
        author: "Walt Whitman",
        description: null,
        alternateDescription: null,
      },
      {
        name: "Cantata",
        title: "Ach wie fluchtig, ach wie nichtig BWV 26",
        author: "JS Bach",
        description: null,
        alternateDescription: null,
      },
      {
        name: "Science",
        title: "Can You Hear The Shape of a Drum?",
        author: "Professor John Chapman",
        description: null,
        alternateDescription: null,
      },
      {
        name: "Closing",
        title: null,
        author: null,
        description: null,
        alternateDescription: null,
      },
      {
        name: "Postlude",
        title: "Darmstadt Ouverture TWV55:g4 'Gasconnade'",
        author: "Telemann",
        description: null,
        alternateDescription: null,
      },
    ],
    theme: {
      fontType: 2,
      primaryColour: "#FA825D",
      textColour: "#FFFFFF",
      highlightColour: "#FFFFC7",
    },
  },
];
