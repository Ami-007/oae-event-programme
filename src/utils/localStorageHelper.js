export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
