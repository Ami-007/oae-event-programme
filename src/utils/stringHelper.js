export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
