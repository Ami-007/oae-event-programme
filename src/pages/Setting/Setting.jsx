import React, { useState, useEffect } from "react";
import Layout from "../../components/Layout/Layout";
import { useParams } from "react-router-dom";
import { getEvent } from "../../assets/data/api";
import LoadingSpinner from "../../components/LoadingSpinner/LoadingSpinner";
import { getDarkTheme, setDarkTheme } from "../../utils/localStorageHelper";
import ErrorPage from "../ErrorPage/ErrorPage";

const Setting = () => {
  const { eventId } = useParams();
  const [event, setEvent] = useState();

  useEffect(async () => {
    setEvent((await getEvent(eventId)) ?? null);
  }, [getEvent, eventId, setEvent]);

  if (event === undefined) return <LoadingSpinner />;

  if (event === null) return <ErrorPage />;

  const { theme } = event;

  const handleChange = (e) => {
    let checked = e.target.checked;
    setDarkTheme(checked);
    window.location.reload();
  };

  const isDarkMode = getDarkTheme();

  return (
    <Layout
      eventId={eventId}
      fontType={theme.fontType}
      themeType={theme.primaryColour}
      textColor={theme.textColour}
      highlightColor={theme.highlightColour}
    >
      <input
        type="checkbox"
        id="dark-mode"
        onChange={handleChange}
        checked={isDarkMode}
      />
      <label htmlFor="dark-mode">Dark Mode</label>
    </Layout>
  );
};

export default Setting;
