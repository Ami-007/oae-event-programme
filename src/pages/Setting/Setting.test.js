import { customRender } from "../../utils/testUtils";
import Setting from "./Setting";
import { screen } from "@testing-library/react";
import { getMockEventById } from "../../assets/data/data";

const mockEvent = getMockEventById("6bb8ffe8-6e5e-4abc-be7f-b05b749849cf");
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: () => [mockEvent, jest.fn()],
}));

test("renders setting component", () => {
  const { container } = customRender(<Setting />);
  expect(container).toMatchSnapshot();
});

test("renders setting component", () => {
  customRender(<Setting />);
  const inputElement = screen.getByRole("checkbox");
  expect(inputElement).toBeInTheDocument();
});
