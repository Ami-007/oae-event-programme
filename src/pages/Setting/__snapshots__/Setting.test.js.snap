// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders setting component 1`] = `
<div>
  <div
    class="layout modern"
  >
    <div>
      <input
        id="dark-mode"
        type="checkbox"
      />
      <label
        for="dark-mode"
      >
        Dark Mode
      </label>
    </div>
    <nav
      class="navbar"
    >
      <div
        class="navbar__content"
      >
        <a
          class="navbar__nav-link "
          href="/undefined/event-schedule"
        >
          <svg
            height="36px"
            viewBox="0 0 16 16"
            width="36px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 13c0 1.105-1.12 2-2.5 2S7 14.105 7 13s1.12-2 2.5-2 2.5.895 2.5 2z"
            />
            <path
              d="M12 3v10h-1V3h1z"
              fill-rule="evenodd"
            />
            <path
              d="M11 2.82a1 1 0 0 1 .804-.98l3-.6A1 1 0 0 1 16 2.22V4l-5 1V2.82z"
            />
            <path
              d="M0 11.5a.5.5 0 0 1 .5-.5H4a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5zm0-4A.5.5 0 0 1 .5 7H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5zm0-4A.5.5 0 0 1 .5 3H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="navbar__title"
          >
            SCHEDULE
          </span>
        </a>
      </div>
      <div
        class="navbar__content"
      >
        <a
          class="navbar__nav-link "
          href="/undefined/home"
        >
          <svg
            height="36px"
            viewBox="0 0 32 32"
            width="36px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M 16 2.59375 L 15.28125 3.28125 L 2.28125 16.28125 L 3.71875 17.71875 L 5 16.4375 L 5 28 L 14 28 L 14 18 L 18 18 L 18 28 L 27 28 L 27 16.4375 L 28.28125 17.71875 L 29.71875 16.28125 L 16.71875 3.28125 Z M 16 5.4375 L 25 14.4375 L 25 26 L 20 26 L 20 16 L 12 16 L 12 26 L 7 26 L 7 14.4375 Z"
            />
          </svg>
          <span
            class="navbar__title"
          >
            HOME
          </span>
        </a>
      </div>
      <div
        class="navbar__content"
      >
        <a
          class="navbar__nav-link "
          href="/undefined/our-story"
        >
          <svg
            height="36px"
            viewBox="0 0 50 50"
            width="36px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M 25 2 C 12.309295 2 2 12.309295 2 25 C 2 37.690705 12.309295 48 25 48 C 37.690705 48 48 37.690705 48 25 C 48 12.309295 37.690705 2 25 2 z M 25 4 C 36.609824 4 46 13.390176 46 25 C 46 36.609824 36.609824 46 25 46 C 13.390176 46 4 36.609824 4 25 C 4 13.390176 13.390176 4 25 4 z M 25 11 A 3 3 0 0 0 22 14 A 3 3 0 0 0 25 17 A 3 3 0 0 0 28 14 A 3 3 0 0 0 25 11 z M 21 21 L 21 23 L 22 23 L 23 23 L 23 36 L 22 36 L 21 36 L 21 38 L 22 38 L 23 38 L 27 38 L 28 38 L 29 38 L 29 36 L 28 36 L 27 36 L 27 21 L 26 21 L 22 21 L 21 21 z"
            />
          </svg>
          <span
            class="navbar__title"
          >
            ABOUT
          </span>
        </a>
      </div>
    </nav>
  </div>
</div>
`;
